ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,896 @@
-
-    * `transform` ["identity", "normalize", "onehot" optional]:
-        - For `Categorical` dimensions, the following transformations are
-          supported.
-
-          - "onehot" (default) one-hot transformation of the original space.
-          - "identity" same as the original space.
-
-        - For `Real` and `Integer` dimensions, the following transformations
-          are supported.
-
-          - "identity", (default) the transformed space is the same as the
-            original space.
-          - "normalize", the transformed space is scaled to be between 0 and 1.
-
-    Returns
-    -------
-    * `dimension`:
-        Dimension instance.
-    """
-    if isinstance(dimension, Dimension):
-        return dimension
-
-    if not isinstance(dimension, (list, tuple, np.ndarray)):
-        raise ValueError("Dimension has to be a list or tuple.")
-
-    # A `Dimension` described by a single value is assumed to be
-    # a `Categorical` dimension. This can be used in `BayesSearchCV`
-    # to define subspaces that fix one value, e.g. to choose the
-    # model type, see "sklearn-gridsearchcv-replacement.ipynb"
-    # for examples.
-    if len(dimension) == 1:
-        return Categorical(dimension, transform=transform)
-
-    if len(dimension) == 2:
-        if any([isinstance(d, (str, bool)) or isinstance(d, np.bool_)
-                for d in dimension]):
-            return Categorical(dimension, transform=transform)
-        elif all([isinstance(dim, numbers.Integral) for dim in dimension]):
-            return Integer(*dimension, transform=transform)
-        elif any([isinstance(dim, numbers.Real) for dim in dimension]):
-            return Real(*dimension, transform=transform)
-        else:
-            raise ValueError("Invalid dimension {}. Read the documentation for"
-                             " supported types.".format(dimension))
-
-    if len(dimension) == 3:
-        if (any([isinstance(dim, (float, int)) for dim in dimension[:2]]) and
-                dimension[2] in ["uniform", "log-uniform"]):
-            return Real(*dimension, transform=transform)
-        else:
-            return Categorical(dimension, transform=transform)
-
-    if len(dimension) > 3:
-        return Categorical(dimension, transform=transform)
-
-    raise ValueError("Invalid dimension {}. Read the documentation for "
-                     "supported types.".format(dimension))
-
-
-class Dimension(object):
-    """Base class for search space dimensions."""
-
-    prior = None
-
-    def rvs(self, n_samples=1, random_state=None):
-        """Draw random samples.
-
-        Parameters
-        ----------
-        * `n_samples` [int or None]:
-            The number of samples to be drawn.
-
-        * `random_state` [int, RandomState instance, or None (default)]:
-            Set random state to something other than None for reproducible
-            results.
-        """
-        rng = check_random_state(random_state)

```

Comment:
As discussed in the comments, I'd suggest keeping the explicit assignment of an RNG here, but instead of using `check_random_state` and so on, instead change the function call so the third input is called `seed` to encourage this use, and still allow it to accept int, RandomState or None.

Then, in the generation of `index_array` you can then call `rng.uniform(size=n_samples)`, which should work no matter if the user passes a seed number or a RandomState.